src/admin/index.js

```diff
@@ -173,6 +173,25 @@ module.exports = ({
     })
   }
 
+  /**
+   * @param {string} topic
+   */
+
+  const fetchTopicOffsets = async topic => {
+    if (!topic || typeof topic !== 'string') {
+      throw new KafkaJSNonRetriableError(`Invalid topic ${topic}`)
+    }
+
+    const topicPartitions = await findTopicPartitions(cluster, topic)
+    const partitions = topicPartitions.map(partition => ({ partition }))
+    const broker = await cluster.findControllerBroker()
+    const offsets = await broker.listOffsets({ topics: [{ topic, partitions }] })
+
+    return offsets.responses
+      .find(response => response.topic === topic)
```

Comment:
What happens if the topic is not found by the broker? Are we guaranteed to get back a list containing the topic we asked for? I'm fine with us throwing an error, as long as it's not `TypeError: Cannot read property 'partitions' of undefined`.

Reply:
@Nevon good question i was blindly following [fetchOffsets](https://github.com/tulios/kafkajs/blob/master/src/admin/index.js#L181) as example i guess it suffers from the same issue

Reply:
Not really, because there it filters the array instead of using `find`, which means that even if the topic is not found, it still returns an array, which means that the `map` will always work, even if it just results in an empty array.

Reply:
@Nevon not sure how this would make difference  tbh if topic is not found or don't exist `findTopicPartitions`  will throw error anyways ` This server does not host this topic-partition` this also apply to ` fetchOffsets`
as long as topic exist we should be good, topic will have at least 1 partition and it will satisfy arguments for `listOffsets` means we should always have `responses` with at least one partition ? unless `listOffsets` returns `[]` responses for what ever reason

Reply:
That's exactly what I wasn't sure of. If the methods will throw if the topic doesn't exist or can't be found, it's all good. I just wanted to make sure that we don't end up with a TypeError because of an invalid assumption. 👍